### ISSUE TITLE
openssl3 - what is the difference between EVP_PKEY_keygen and EVP_PKEY_generate?

### ISSUE THREAD

**Issue by johnvenn** [openssl/openssl#21161]
in openssl3.0, you introduced a new API EVP_PKEY_generate.

In my old application using openssl1.1, I still use the API EVP_PKEY_keygen to generate 4096 bits RSA key. 

what is the difference between the 2 APIs ?
how to decide which one to use ?

**Comment by t8m** [comment]
The EVP_PKEY_generate is just a simplified API built upon the EVP_PKEY_keygen. Both are supported and equivalent. Choose the one which suits you better.

**Comment by t8m** [comment]
Actually I mistook EVP_PKEY_generate with EVP_PKEY_Q_keygen.

For EVP_PKEY_generate the answer is in the manual page:

EVP_PKEY_paramgen() and EVP_PKEY_keygen() do exactly the same thing as
EVP_PKEY_generate(), after checking that the corresponding EVP_PKEY_paramgen_init()
or EVP_PKEY_keygen_init() was used to initialize I<ctx>.
These are older functions that are kept for backward compatibility.
It is safe to use EVP_PKEY_generate() instead.

So basically no difference at all.

**Comment by johnvenn** [comment]
thanks for the clear explanation. appreciated.

**Comment by johnvenn** [comment]
resolved
